app/components/nextflow/samplesheet/header_component.rb

```diff
@@ -16,17 +17,27 @@ def initialize(namespace_id:, header:, property:, samples:, metadata_fields:, re
         @metadata_fields = metadata_fields
         @required_properties = required_properties
         @workflow_params = workflow_params
+        @selected = default_header
       end
 
       # rubocop:enable Metrics/ParameterLists
 
       private
 
-      def metadata_fields_for_field(field)
+      def metadata_fields_for_field
```

Comment:
Suggestion (blocking): can you move this into the `public` section since the view is calling this method.

Or actually might be better if you leave it here and instead call the method from `initialize` and assign it to a variable.